### PR TITLE
Style, organization changes (no more oauth)

### PR DESCRIPTION
**Pull request by alg1142**

### CODE REVIEW

File: server/api/routes/api.js [hunk 1]
```diff
@@ -29,21 +19,9 @@ router.get(
...
-      logger.debug(`ES query failed: ${e.message}`)
-
-      const errorCode = e.meta.statusCode
-      res.status(errorCode)
-
-      if (errorCode === 404) return res.json({error: 'Not found'})
-      else return res.json({error: 'Unknown error'})
-    }
+    const {logs, statusCode} = yield search(req.query)
+    res.status(statusCode)
```
Comment (johnpyp) [inline server/api/routes/api.js]: Can be simplified to `return res.status(statusCode).json(logs)`

Comment (alg1142) [reply server/api/routes/api.js]: one thing per line

File: client/src/views/About.vue [hunk 2]
```diff
@@ -10,14 +10,41 @@
       <p>
         contact me at rustlesearch@gmail.com for inquiry
       </p>
+      <q-btn
+        color="primary"
+        @click="signUp"
+      >
+        Signup with google
```
Comment (alg1142) [inline client/src/views/About.vue]: we dont use google

File: server/api/lib/auth.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import sa from 'superagent'
+import jwt from 'jsonwebtoken'
+import logger from '@lib/logger'
+export const jwtToPatreonMiddleware = async(req, res, next) => {
```
Comment (alg1142) [inline server/api/lib/auth.js]: middleware goes in @ middleware

Comment (alg1142) [inline server/api/lib/auth.js]: async is banned in this repo

File: server/api/lib/auth.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import sa from 'superagent'
```
Comment (alg1142) [inline server/api/lib/auth.js]: what the fuck is this naming

Comment (alg1142) [reply server/api/lib/auth.js]: why do you need another request lib, just use request-promise

File: server/api/lib/auth.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    logger.error(e)
+    return res
+      .status(400)
+      .json({error: 'Patreon fetch info failed', revalidate: false})
+  }
+}
+
+export const toJwt = accessToken => {
+  const token = jwt.sign({accessToken}, process.env.KEY_SECRET, {
+    expiresIn: '15m',
```
Comment (alg1142) [inline server/api/lib/auth.js]: why is this so short

you should also consider using jwt-simple

File: server/api/lib/auth.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  if (decoded === null)
+    return res.status(401).json({error: 'Jwt invalid', revalidate: true})
+  try {
+    const {body} = await sa
+      .get('https://www.patreon.com/api/oauth2/api/current_user')
+      .set('authorization', `Bearer ${decoded.accessToken}`)
+    req.patreon = body
+    return next()
+  } catch (e) {
+    logger.error(e)
```
Comment (alg1142) [inline server/api/lib/auth.js]: this isnt a real error, debug

File: server/api/lib/auth.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  })
+  return token
+}
+
+export const decodeJwt = token => {
+  try {
+    const decoded = jwt.verify(token, process.env.KEY_SECRET)
+    return decoded
+  } catch (e) {
+    logger.info('Invalid jwt token', e.message, token)
```
Comment (alg1142) [inline server/api/lib/auth.js]: debug

File: server/api/routes/api.js [hunk 8]
```diff
@@ -2,7 +2,8 @@ import {search} from '@lib/elastic'
 import {co, fs} from '@lib/util'
 import express from 'express'
 import ratelimit from '@middleware/rate-limiter'
-
+// import popupTools from 'popup-tools'
```
Comment (alg1142) [inline server/api/routes/api.js]: no commented out deps

File: server/api/routes/api.js [hunk 9]
```diff
@@ -34,4 +35,17 @@ router.get(
   }),
 )
 
+router.get('/callback/patreon', async(req, res) => {
+  const resp = req.session.grant.response
+  if (!resp.access_token) return res.json({error: 'Authentication failed'})
+  return res.json({jwt: toJwt(resp.access_token)})
+  // res.end(popupTools.popupResponse(profile))
+})
```
Comment (alg1142) [inline server/api/routes/api.js]: put a space between blocks

File: server/api/routes/api.js [hunk 10]
```diff
@@ -34,4 +35,17 @@ router.get(
   }),
 )
 
+router.get('/callback/patreon', async(req, res) => {
```
Comment (alg1142) [inline server/api/routes/api.js]: why is this async

File: server/api/routes/api.js [hunk 11]
```diff
@@ -34,4 +35,17 @@ router.get(
...
+  const resp = req.session.grant.response
+  if (!resp.access_token) return res.json({error: 'Authentication failed'})
+  return res.json({jwt: toJwt(resp.access_token)})
+  // res.end(popupTools.popupResponse(profile))
```
Comment (alg1142) [inline server/api/routes/api.js]: no comments

File: server/api/server.js [hunk 12]
```diff
@@ -49,6 +48,22 @@ app.use(
...
+  grant({
+    defaults: {
+      protocol: 'https',
+      host: `api.${process.env.DOMAIN}`,
+      transport: 'session',
+    },
+    patreon: {
+      key: process.env.PATREON_CLIENT_ID,
+      secret: process.env.PATREON_CLIENT_SECRET,
+      // scope: ['pledges-to-me'],
```
Comment (alg1142) [inline server/api/server.js]: no comments

File: server/api/server.js [hunk 13]
```diff
@@ -49,6 +48,22 @@ app.use(
   }),
 )
 
+app.use(
+  grant({
+    defaults: {
+      protocol: 'https',
+      host: `api.${process.env.DOMAIN}`,
```
Comment (alg1142) [inline server/api/server.js]: no api prefix. this is hardcoding, make a full var for this

File: server/docker-compose.yml [hunk 14]
```diff
@@ -16,7 +16,23 @@ services:
...
+    image: node:12
+    volumes:
+      - .:/node_app
+    working_dir: /node_app
+    command: echo 1
+    environment:
+      - VIRTUAL_HOST=api.${DOMAIN}
+      - VIRTUAL_PORT=${APP_PORT}
+      - LETSENCRYPT_HOST=api.${DOMAIN}
+      - RESOLVERS="1.1.1.1 1.0.0.1 valid=300s"
```
Comment (alg1142) [inline server/docker-compose.yml]: you should make the resolvers global in the tmpl instead and remove them from here

File: server/docker-compose.yml [hunk 15]
```diff
@@ -16,7 +16,23 @@ services:
...
+      - SSL_POLICY=Mozilla-Modern
```
Comment (alg1142) [inline server/docker-compose.yml]: this does nothing

File: server/docker-compose.yml [hunk 16]
```diff
@@ -16,7 +16,23 @@ services:
...
+  rustlesearch_dev:
+    image: node:12
+    volumes:
+      - .:/node_app
+    working_dir: /node_app
+    command: echo 1
+    environment:
+      - VIRTUAL_HOST=api.${DOMAIN}
+      - VIRTUAL_PORT=${APP_PORT}
+      - LETSENCRYPT_HOST=api.${DOMAIN}
```
Comment (alg1142) [inline server/docker-compose.yml]: another hardcoded api.

File: server/package.json [hunk 17]
```diff
@@ -35,24 +35,32 @@
   ],
   "dependencies": {
     "@elastic/elasticsearch": "7.1.0",
+    "axios": "0.19.0",
```
Comment (alg1142) [inline server/package.json]: not used

File: server/package.json [hunk 18]
```diff
@@ -35,24 +35,32 @@
...
     "bluebird": "3.5.5",
+    "body-parser": "1.19.0",
```
Comment (alg1142) [inline server/package.json]: not used

File: server/package.json [hunk 19]
```diff
@@ -35,24 +35,32 @@
...
+    "grant": "4.5.2",
+    "grant-express": "4.5.2",
     "helmet": "3.18.0",
     "ioredis": "4.10.0",
+    "jsonwebtoken": "8.5.1",
     "lodash": "4.17.11",
     "luxon": "1.16.0",
     "node-etl": "1.0.3",
     "pino": "5.12.6",
+    "popup-tools": "1.0.2",
```
Comment (alg1142) [inline server/package.json]: not used

File: server/package.json [hunk 20]
```diff
@@ -35,24 +35,32 @@
...
+    "querystring": "0.2.0",
     "ramda": "0.26.1",
     "rate-limiter-flexible": "1.0.2",
     "redis-mock": "0.44.0",
-    "request": "2.88.0"
+    "request": "2.88.0",
+    "superagent": "5.1.0"
```
Comment (alg1142) [inline server/package.json]: not needed

File: server/package.json [hunk 21]
```diff
@@ -35,24 +35,32 @@
...
+    "grant-express": "4.5.2",
     "helmet": "3.18.0",
     "ioredis": "4.10.0",
+    "jsonwebtoken": "8.5.1",
     "lodash": "4.17.11",
     "luxon": "1.16.0",
     "node-etl": "1.0.3",
     "pino": "5.12.6",
+    "popup-tools": "1.0.2",
+    "querystring": "0.2.0",
```
Comment (alg1142) [inline server/package.json]: not used

File: server/api/server.js [hunk 22]
```diff
@@ -49,6 +48,22 @@ app.use(
...
+      transport: 'session',
+    },
+    patreon: {
+      key: process.env.PATREON_CLIENT_ID,
```
Comment (alg1142) [inline server/api/server.js]: add new vars into the config check

File: server/api/lib/auth.js (hunk 4 above)
Comment (johnpyp) [reply server/api/lib/auth.js]: I like superagent, comfy

File: server/package.json (hunk 19 above)
Comment (johnpyp) [reply server/package.json]: we will use it, soon

File: server/package.json (hunk 20 above)
Comment (johnpyp) [reply server/package.json]: gonna convert to superagent, comfy

File: server/api/lib/auth.js (hunk 5 above)
Comment (johnpyp) [reply server/api/lib/auth.js]: testing purposes, I'll lengthen it

File: server/api/lib/auth.js (hunk 4 above)
Comment (alg1142) [reply server/api/lib/auth.js]: wrong

File: server/package.json (hunk 19 above)
Comment (alg1142) [reply server/package.json]: not in the backend

Comment (alg1142) [reply server/package.json]: this shit is a meme friend

File: server/api/lib/auth.js (hunk 4 above)
Comment (johnpyp) [reply server/api/lib/auth.js]: wrong

File: server/docker-compose.yml [hunk 23]
```diff
@@ -7,11 +7,10 @@ services:
     restart: unless-stopped
     env_file: .env
     environment:
-      - VIRTUAL_HOST=api.${DOMAIN}
+      - VIRTUAL_HOST=${SUBDOMAIN}.${DOMAIN}
```
Comment (alg1142) [inline server/docker-compose.yml]: i still dont like this approach. its possible there is no subdomain... and then .site.com is not a valid domain

**Comment by alg1142** [comment]
also touching on the domain some more.. the letseencrypt container *does not care* about the root domain, only whole paths, so theres just no point to split these

**Pull request merged**